langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```
@@ -0,0 +1,426 @@
+        if (isNullOrEmpty(ids) || isNullOrEmpty(embeddings)) {
+            log.info("Empty embeddings - no ops");
+            return;
+        }
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(),
+                "embeddings size is not equal to embedded size");
+
+        List<Document> searchDocuments = new ArrayList<>();
+        for (int i = 0; i < ids.size(); ++i) {
+            Document document = new Document();
+            document.setId(ids.get(i));
+            document.setContent_vector(floatsArrayToList(embeddings.get(i).vector()));
+            if (embedded != null) {
+                document.setContent(embedded.get(i).text());
+                Document.Metadata metadata = new Document.Metadata();
+                metadata.setSource("langchain4j");
+                List<Document.Metadata.Attribute> attributes = new ArrayList<>();
+                for (Map.Entry<String, String> entry : embedded.get(i).metadata().asMap().entrySet()) {
+                    Document.Metadata.Attribute attribute = new Document.Metadata.Attribute();
+                    attribute.setKey(entry.getKey());
+                    attribute.setValue(entry.getValue());
+                    attributes.add(attribute);
+                }
+                metadata.setAttributes(attributes);
+                document.setMetadata(metadata);
+            } else {
+                document.setContent("");
+            }
+            searchDocuments.add(document);
+        }
+        List<IndexingResult> indexingResults = searchClient.uploadDocuments(searchDocuments).getResults();
+        for (IndexingResult indexingResult : indexingResults) {
+            if (!indexingResult.isSucceeded()) {
+                log.error("Failed to add embedding: {}", indexingResult.getErrorMessage());
```

should we maybe throw an exception here instead of logging?

---

yes